XIVComboPlugin/IconReplacer.cs

```diff
@@ -898,6 +916,145 @@ private ulong GetIconDetour(byte self, uint actionID)
+            {
+                var PCTGauge = JobGauges.Get<PCTGauge>();
+                if (actionID == PCT.LandscapeMotif)
+                {
+                    if (PCTGauge.LandscapeMotifDrawn)
+                        return PCT.StarryMuse;
+                    if (SearchBuffArray(PCT.StarStruck))
+                        return PCT.StarPrism;
+                    return PCT.StarryMotif;
+                }
+            }
+            
+            //VIPER
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.ViperDeathRattleMaximum))
+            {
+                if (actionID == VPR.SteelFangs || actionID == VPR.DreadFangs)
+                {
+                    if (iconHook.Original(self, VPR.SerpentsTail) == VPR.DeathRattle)
+                        return VPR.DeathRattle;
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.ViperDeathRattleMaximum))
+            {
+
+                if (actionID == VPR.DreadMaw || actionID == VPR.SteelMaw)
+                {
+                    if (iconHook.Original(self, VPR.SerpentsTail) == VPR.LastLash)
+                        return VPR.LastLash;
+                }
+            }
+            
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.ViperDeathRattleMaximum)){
+                if (actionID == VPR.Reawaken && SearchBuffArray(VPR.Buffs.Reawakened))
+                    {
+                        var gauge = JobGauges.Get<VPRGauge>();
+
+                        if (level >= VPR.Levels.Legacies)
+                        {
+                            if (iconHook.Original(self, VPR.SerpentsTail) == VPR.FirstLegacy || 
+                                iconHook.Original(self, VPR.SerpentsTail) == VPR.SecondLegacy ||
+                                iconHook.Original(self, VPR.SerpentsTail) == VPR.ThirdLegacy ||
+                                iconHook.Original(self, VPR.SerpentsTail) == VPR.FourthLegacy)
+                                return iconHook.Original(self, VPR.SerpentsTail);
+                        }
+
+                        var maxtribute = 4;
+                        if (level >= VPR.Levels.Ouroboros)
+                            maxtribute = 5;
+                        if (gauge.AnguineTribute == maxtribute)
+                            return VPR.FirstGeneration;
+                        if (gauge.AnguineTribute == maxtribute - 1)
+                            return VPR.SecondGeneration;
+                        if (gauge.AnguineTribute == maxtribute - 2)
+                            return VPR.ThirdGeneration;
+                        if (gauge.AnguineTribute == maxtribute - 3)
+                            return VPR.FourthGeneration;
+                        if (gauge.AnguineTribute == 1 && level >= VPR.Levels.Ouroboros)
+                            return VPR.Ouroboros;
+                    }
+               
+                return iconHook.Original(self, actionID);
```

Comment:
pretty sure this breaks everything underneath it and can be removed without any negative side-effects for this combo

Reply:
If you want an example of what I mean, place any other viper combo beneath this one and it will no longer work

Reply:
Ah nice catch, I added that in testing and missed removing it. Also just noticed I have everything keyed off of the Death Rattle toggle instead of their own. Thanks for getting my eyes back on this!

Reply:
not a problem. I added a few more viper combos for Uncoiled Fury and Hunters'/Swiftskin Coil, but I think its to the point where it probably wouldn't be approved because it kinda deletes all thought from the class xD. Just ran into those issues while adding them